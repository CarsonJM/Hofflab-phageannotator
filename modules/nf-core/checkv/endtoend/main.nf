process CHECKV_ENDTOEND {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/checkv:1.0.1--pyhdfd78af_0':
        'biocontainers/checkv:1.0.1--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(fasta)
    path db

    output:
    tuple val(meta), path ("${prefix}_quality_summary.tsv") , emit: quality_summary
    tuple val(meta), path ("${prefix}_contamination.tsv")   , emit: contamination
    tuple val(meta), path ("${prefix}_completeness.tsv")    , emit: completeness
    tuple val(meta), path ("${prefix}_viruses.fna.gz")      , emit: viruses
    tuple val(meta), path ("${prefix}_proteins.faa.gz")     , emit: proteins
    path "versions.yml"                                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"

    """
    checkv \\
        end_to_end \\
        ${args} \\
        -t ${task.cpus} \\
        -d ${db} \\
        ${fasta} \\
        ${prefix}

    mv ${prefix}/quality_summary.tsv ${prefix}_quality_summary.tsv
    mv ${prefix}/contamination.tsv ${prefix}_contamination.tsv
    mv ${prefix}/completeness.tsv ${prefix}_completeness.tsv
    gzip ${prefix}/*.fna
    cat ${prefix}/*.fna.gz > ${prefix}_viruses.fna.gz
    mv ${prefix}/tmp/proteins.faa ${prefix}_proteins.faa
    gzip ${prefix}_proteins.faa
    rm -rf ./${prefix}/*

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        checkv: \$(checkv -h 2>&1 | sed -n "s/Prophage Tracer V//; s/\s.*//;  2p")
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"

    """
    touch ${prefix}_quality_summary.tsv
    touch ${prefix}_contamination.tsv
    touch ${prefix}_completeness.tsv
    echo "" | gzip > ${prefix}_viruses.fna.gz
    echo "" | gzip > ${prefix}_proteins.faa.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        checkv: \$(checkv -h 2>&1  | sed -n 's/^.*CheckV v//; s/: assessing.*//; 1p')
    END_VERSIONS
    """
}
