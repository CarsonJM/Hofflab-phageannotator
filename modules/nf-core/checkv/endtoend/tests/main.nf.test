nextflow_process {

    name "Test Process CHECKV_ENDTOEND"
    script "../main.nf"
    process "CHECKV_ENDTOEND"


    setup {
        run("CHECKV_DOWNLOADDATABASE") {
            script "../../downloaddatabase/main.nf"
        }
    }


    test("fasta.gz") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test1' ],
                    file("${projectDir}/assets/test-datasets/fastas/lambda.fasta.gz", checkIfExists: true)
                ]
                input[1] = CHECKV_DOWNLOADDATABASE.out.checkv_db
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.quality_summary,
                    process.out.contamination,
                    file(process.out.completeness[0][1]).name,
                    process.out.viruses,
                    process.out.proteins,
                    process.out.versions
                    ).match()
                }
            )
        }
    }


    test("fasta.gz - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'test1' ],
                    file("${projectDir}/assets/test-datasets/fastas/lambda.fasta.gz", checkIfExists: true)
                ]
                input[1] = CHECKV_DOWNLOADDATABASE.out.checkv_db
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
            )
        }
    }
}
