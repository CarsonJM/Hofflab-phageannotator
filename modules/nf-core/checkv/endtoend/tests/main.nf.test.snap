{
    "fasta.gz - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_quality_summary.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_contamination.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_completeness.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_viruses.fna.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_proteins.faa.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "5": [
                    "versions.yml:md5,e2b3c1c7aa54062a80b97e3c70bbb28a"
                ],
                "completeness": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_completeness.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "contamination": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_contamination.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "proteins": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_proteins.faa.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "quality_summary": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_quality_summary.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e2b3c1c7aa54062a80b97e3c70bbb28a"
                ],
                "viruses": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_viruses.fna.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-24T14:00:07.284547244"
    },
    "fasta.gz": {
        "content": [
            [
                [
                    {
                        "id": "test1"
                    },
                    "test1_quality_summary.tsv:md5,01e7069e0c3b3d36b4d8eb02845b3086"
                ]
            ],
            [
                [
                    {
                        "id": "test1"
                    },
                    "test1_contamination.tsv:md5,aff5543ad5c34d1adae45336c07d7364"
                ]
            ],
            "test1_completeness.tsv",
            [
                [
                    {
                        "id": "test1"
                    },
                    "test1_viruses.fna.gz:md5,c9cebc9db75d3fd8cd58e5829cecd8b4"
                ]
            ],
            [
                [
                    {
                        "id": "test1"
                    },
                    "test1_proteins.faa.gz:md5,e0c10ef5758f5472b28a2b71071c3ce2"
                ]
            ],
            [
                "versions.yml:md5,e14433c20aa8085bf627b8ae97dde94e"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-24T13:31:36.034984427"
    }
}